Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -63,23 +65,67 @@ - (instancetype)initWithRecipientId:(NSString *)recipientId
     return self;
 }
 
+- (nullable NSString *)profileName
+{
+    @synchronized(self)
+    {
+        return _profileName;
+    }
+}
+
 - (void)setProfileName:(nullable NSString *)profileName
 {
-    _profileName = [profileName ows_stripped];
+    @synchronized(self)
+    {
+        _profileName = [profileName ows_stripped];
+    }
+}
+
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(nullable NSZone *)zone
+{
+    UserProfile *copy = [[UserProfile alloc] initWithRecipientId:self.recipientId];
+    copy.profileKey = self.profileKey;
+    copy.profileName = self.profileName;
+    copy.avatarUrlPath = self.avatarUrlPath;
+    copy.avatarFileName = self.avatarFileName;
+    copy.lastUpdateDate = self.lastUpdateDate;
+    return copy;
 }
 
 #pragma mark - NSObject
 
-- (BOOL)isEqual:(UserProfile *)other
+- (BOOL)areEqualStrings:(NSString *)left other:(NSString *)right
```

Comment:
nit: looks like params should be marked `nullable`

Reply:
Removed.